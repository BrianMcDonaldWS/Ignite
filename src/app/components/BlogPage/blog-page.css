.blog {
  margin-top: -7.5rem !important;
}

.Page {
  padding-left: 45px;
  padding-right: 45px;
}
