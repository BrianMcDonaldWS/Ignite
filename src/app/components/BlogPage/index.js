import React from 'react';
import PropTypes from 'prop-types';
import makeClass from 'classnames';
import ReactRouterPropTypes from 'react-router-prop-types';

import BlogHero from '../BlogHero';
import styles from './blog-page.css';

const BlogPage = ({ Page, location, plugins, blogHero, className }) => (
  <div>
    <BlogHero key="hero" location={location} blogHero={blogHero} />
    <div className={makeClass(className, 'columns', styles.blog)}>
      <div
        className={makeClass(
          'column',
          'content',
          'is-two-thirds-tablet',
          'is-three-quarters-desktop'
        )}
      >
        <Page plugins={plugins} className={styles.Page} />
      </div>
    </div>
  </div>
);

BlogPage.propTypes = {
  Page: PropTypes.node.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  plugins: PropTypes.array,
  blogHero: PropTypes.string,
  className: PropTypes.string
};

BlogPage.defaultProps = {
  plugins: [],
  blogHero: null,
  className: null
};

export default BlogPage;
