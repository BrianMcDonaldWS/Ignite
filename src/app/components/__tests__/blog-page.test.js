import React from 'react';

import BlogPage from '../BlogPage';
import renderToJson from './utils/render-to-json';

test('BlogPage', () => {
  expect(renderToJson(<BlogPage Page={() => {}} />)).toMatchSnapshot();
});
