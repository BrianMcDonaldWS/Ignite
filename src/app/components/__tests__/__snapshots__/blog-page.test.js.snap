// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage 1`] = `
<div>
  <BlogHero
    blogHero={null}
  />
  <div
    className="columns blog"
  >
    <div
      className="column content is-two-thirds-tablet is-three-quarters-desktop"
    >
      <Page
        className="Page"
        plugins={Array []}
      />
    </div>
  </div>
</div>
`;
